backend/src/services/service-token/service-token-service.ts

```diff
@@ -185,11 +189,39 @@ export const serviceTokenServiceFactory = ({
     return { ...serviceToken, lastUsed: new Date(), orgId: project.orgId };
   };
 
+  const notifyExpiredTokens = async () => {
+    const appCfg = getConfig();
+
+    const expiredTokens = await serviceTokenDAL.findExpiredTokens();
+    if (expiredTokens.length === 0) return;
+
+    await Promise.all(
+      expiredTokens.map(async (token) => {
+        try {
+          await smtpService.sendMail({
+            recipients: [token.createdByEmail],
+            subjectLine: "Service Token Expired",
+            template: SmtpTemplates.ServiceTokenExpired,
+            substitutions: {
+              tokenName: token.name,
+              projectName: token.projectName,
+              url: `${appCfg.SITE_URL}/secret-manager/${token.projectId}/access-management?selectedTab=service-tokens`
+            }
+          });
+          await serviceTokenDAL.update({ id: token.id }, { notificationSent: true });
+        } catch (error) {
+          logger.error(error, `Failed to send expiration notification for token ${token.id}:`);
+        }
+      })
+    );
+  };
```

Comment:
note that Infisical is typically deployed in multiple containers/replicas.. this means that